src/spreadinterp.cpp

```diff
@@ -2160,13 +2163,13 @@ FINUFFT_EXPORT int FINUFFT_CDECL setup_spreader(finufft_spread_opts &opts, T eps
               upsampfac);
       return FINUFFT_ERR_HORNER_WRONG_BETA;
     }
-    if (upsampfac <= 1.0) { // no digits would result
-      fprintf(stderr, "FINUFFT setup_spreader: error, upsampfac=%.3g is <=1.0\n",
+    if (upsampfac <= 1.0) { // no digits would result, ns infinite
+      fprintf(stderr, "FINUFFT setup_spreader: error, upsampfac=%.3g is <= 1.0\n",
               upsampfac);
       return FINUFFT_ERR_UPSAMPFAC_TOO_SMALL;
     }
-    // calling routine must abort on above errors, since opts is garbage!
-    if (showwarn && upsampfac > 4.0)
+    // calling routine must abort on above errors, since (spread)opts is garbage!
```

Comment:
we should update the comment to match the code. This is garbage only for a NUFFT

Reply:
I stand by the validity of my comment here. The "above errors" still would require abort, even for SI-only.